Reader Detail Updates (Post Header)

ashiagr opened the pull request:
Task: #12900

This PR refactors post details header view with following changes:

Child PRs:

PR #13060 - Kotlin conversion
PR #13061 - `uiState` extracted for blog section and follow button

- Post header view updated using `uiState` extracted in PR #13061 03db5a1
- Post title moved from `reader_include_post_detail_content.xml` 8790672
- Separated blog post author 8790672

**Merge instructions**

1. Make sure #13060, #13061 are merged
2. Change branch to "issue/12900-rip3-detail-updates-main"
3. Remove "Not Ready for Merge"
4. Merge this PR

#### To test

- That UI elements are more or less in the same position as per [Figma design](https://www.figma.com/file/yG8ieha0yESnr8wCtplr29/Post-Detail-View?node-id=8%3A477)
- Clicking blog section opens blog preview screen

|Post Details Header|
|-------------------|
|<img width="320" alt="Screenshot 2020-10-01 at 10 59 59 AM" src="https://user-images.githubusercontent.com/1405144/94771669-6c033900-03d5-11eb-8b01-bef04887d3fb.png">|

**Note**
- Follow action refactoring and UI polishing will be done in separate PRs.
- ~Most of the changes are due to extraction and reuse of blog section from `ReaderPostUiState` (2c6fac5, 8dbc4f8). I can split this PR into smaller ones if anyone finds it difficult to review.~ - #13060,  #13061 split from this PR


PR submission checklist:

- [ ] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ashiagr committed:
Update ReaderPostDetailsHeaderView using uiState

ashiagr committed:
Display title, author name in header view

ashiagr committed:
Add UiStateBuilder to create post details header view uiState

ashiagr committed:
Remove unused resources

ashiagr committed:
Rename blog section

ashiagr committed:
Merge branch 'issue/12900-rip3-details-updates-part-i' into issue/12900-rip3-details-updates-part-ii

ashiagr committed:
Merge branch 'issue/12900-rip3-details-updates-part-ii-split-kotlin' into issue/12900-rip3-details-updates-part-ii

# Conflicts:
#	WordPress/src/main/res/values/dimens.xml

ashiagr committed:
Merge branch 'issue/12900-rip3-details-updates-part-ii-split-uistate' into issue/12900-rip3-details-updates-part-ii

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt

malinajirka commented on WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt:
```diff
@@ -1522,6 +1517,65 @@ class ReaderPostDetailFragment : ViewPagerFragment(),
+                    post.isFollowedByCurrentUser = isAskingToFollow
+                } else {
+                    val errResId = if (isAskingToFollow) {
+                        string.reader_toast_err_follow_blog
```
NitPick: Lint sometime fails because it marks resources which are missing the "R." prefix as unused. We might want to consider adding the prefix and removing the import. Wdyt?

ashiagr replied:
Fixed in c6acf92

malinajirka commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -1320,6 +1320,7 @@
     <string name="reader_expandable_tags_view_overflow_indicator_expand_title" translatable="false">%1$s+</string>
     <string name="reader_expandable_tags_view_overflow_indicator_collapse_title">Hide</string>
     <string name="reader_discover_interests_header">You might like</string>
+    <string name="reader_post_details_header_by">By</string>
```
We might want to consider renaming this to something like "reader_post_details_header_by_author_name". "By" can have multiple translations in different languages so I think it'd be good to add the context. Wdyt?

ashiagr replied:
Fixed in 82e4378

malinajirka reviewed (commented):
Thanks @ashiagr! All three PRs look good overall. I've encountered just one UI glitch.

When the site title is empty, the header positioning breaks.
Example Post: https://wp.me/pVHyx-t4
It might be related to one of the child PRs - I've tested all the changes altogether in this PR.

<img src="https://user-images.githubusercontent.com/2261188/95215711-939d3b80-07f1-11eb-8319-b77cce5fc3bf.png" width="250px" />

<img src="https://user-images.githubusercontent.com/2261188/95215729-97c95900-07f1-11eb-9516-500a2c82c76f.png" width="250px" />

ashiagr committed:
Add "R." prefix for strings

ashiagr committed:
Rename "By" string key for better context

ashiagr commented:
Thanks @malinajirka ! I've addressed PR review comments. 

> When the site title is empty, the header positioning breaks.

This issue exists on develop as well. I'll create a separate PR targeting develop.
